src/ir/auth_logic/ast.h

```diff
@@ -225,15 +243,15 @@ class ConditionalAssertion {
   }
 
   // A potentially ugly print of the state in this class
-  // for debugging/testing only
-  std::string DebugPrint() const {
-    std::vector<std::string> rhs_strings;
-    rhs_strings.reserve(rhs_.size());
-    for (const BaseFact& base_fact : rhs_) {
-      rhs_strings.push_back(base_fact.DebugPrint());
-    }
-    return absl::StrCat(lhs_.DebugPrint(), ":-",
-                        absl::StrJoin(rhs_strings, ", "));
+  // for debugging only
+  std::string DebugPrint() const;
+
+  bool operator==(const ConditionalAssertion& rhs) const {
+    return lhs_ == rhs.lhs() && rhs_ == rhs.rhs();
+  }
```

Comment:
Might be a better idea to refer to the fields directly: 
```suggestion
  bool operator==(const ConditionalAssertion& other) const {
    return lhs_ == other.lhs_ && rhs_ == other.rhs_;
  }
```


Can you change the other places also?

Reply:
fixed